File: Practica 4/CivitasJava4/CivitasJava/src/civitasjava/CasillaImpuesto.java

package civitasjava;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {
        
    private float importe;                 

    // Construye casillas de tipo IMPUESTO
    CasillaImpuesto(float cantidad, String nombre){
        super(nombre);
        this.init();
        
        this.importe = cantidad;
    }    
        
    /**
     *  @brief Inicia los atributos de instancia del objeto. 
     */
    private void init(){
        this.importe          = 0;           
    }
    
    /**
     * @brief El jugador dado(si existe), paga un impesto
     * @param actual, el indice del jugador
     * @param todos, el vector de jugadores
     * @return void
     */
    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if( jugadorCorrecto(iactual, todos) ){
            super.recibeJugador(iactual, todos);
            
            todos.get(iactual).pagaImpuesto(importe);
        }
    }
}

File: Practica 4/CivitasRuby4/CivitasRuby/lib/casilla_impuesto.rb
module Civitas
  
class CasillaImpuesto < Casilla 
# @brief Constructor de la clase. Casilla impuesto
# @param cantidad, La cantidad a pagar
# @param nombre, el nombre de la casilla
  def initialize(cantidad, nombre)
    super(nombre)
    @importe = cantidad
  end

# @brief El jugador dado(si existe), paga un impesto
# @param actual, el indice del jugador
# @param todos, el vector de jugador
# @return Es void
def recibe_jugador(actual,todos)
  if(jugador_correcto(actual,todos))
    super(actual,todos)
    todos[actual].paga_impuesto(@importe)
  end
end
  
end #fin clase
end #fin modulo
